pkg/api/vlabs/types.go

```diff
@@ -118,6 +118,14 @@ const (
 // OrchestratorProfile contains Orchestrator properties
 type OrchestratorProfile struct {
 	OrchestratorType OrchestratorType `json:"orchestratorType"`
+	KubernetesConfig KubernetesConfig `json:"kubernetesConfig,omitempty"`
+}
+
+// KubernetesConfig contains the Kubernetes config structure, containing
+// Kubernetes specific configuration
+type KubernetesConfig struct {
```

Comment:
**Option 1: OrchestratorOptions  will have both Sovereign cloud config and other options.**

type OrchestratorProfile struct { 
 	OrchestratorType OrchestratorType `json:"orchestratorType"` 
	orchestratorOption[] OrchestratorOptions  `json:"orchestratorOptions,omitempty"` 
} 

type OrchestratorOption struct {
    Key   string
    Value string
}

**Option 2: This option will have separate Sovereign cloud property and other config options will be a separate field. I am leaning towards this option. 

I recommend renaming KubernetesConfig to KubernetesSovereignConfig or something similar so its clear what the differences between both fields would be.**

type OrchestratorProfile struct { 
 	OrchestratorType OrchestratorType `json:"orchestratorType"` 
	KubernetesConfig KubernetesConfig `json:"kubernetesConfig,omitempty"` 
	orchestratorOption[] OrchestratorOptions  `json:"orchestratorOptions,omitempty"` 
} 

type OrchestratorOption struct {
    Key   string
    Value string
}